File: app/src/main/java/ProjectClasses/Covid19ApiSingleton.java
package ProjectClasses;

import Interfaces.CovidApi;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class Covid19ApiSingleton {

    private static Retrofit retrofit = null;
    private static String BASE_URL = "https://api.covid19api.com/";

    public static CovidApi getService(){
        if(retrofit == null){
            retrofit = new Retrofit.Builder()
                    .baseUrl(BASE_URL)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();

        }
        return retrofit.create(CovidApi.class);
    }
}

File: app/src/main/java/com/gian/stayinformed/service/Covid19ApiSingleton.kt
package com.gian.stayinformed.service

import com.gian.stayinformed.interfaces.CovidApi
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

class Covid19ApiSingleton {
    private var retrofit: Retrofit? = null
    private val BASE_URL = "https://api.covid19api.com/"

    fun getService(): CovidApi? {
        if (retrofit == null) {
            retrofit = Retrofit.Builder()
                    .baseUrl(BASE_URL)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build()
        }
        return retrofit!!.create(CovidApi::class.java)
    }
}